_episodes/07-github.md

```diff
@@ -190,13 +197,18 @@ Our local and remote repositories are now in this state:
 
 ![GitHub Repository After First Push](../fig/github-repo-after-first-push.svg)
 
-> ## The '-u' Flag
+> ## Upstream branch
+>
+> You may occasionally see `git push` with an additional `-u` flag, as in:
+>
+> ~~~
+> git push -u origin master
+> ~~~
+> {: .language-bash}
 >
-> You may see a `-u` option used with `git push` in some documentation.  This
-> option is synonymous with the `--set-upstream-to` option for the `git branch`
-> command, and is used to associate the current branch with a remote branch so
-> that the `git pull` command can be used without any arguments. To do this,
-> simply use `git push -u origin master` once the remote has been set up.
+> This option instructs Git to send branch `master` to the remote `origin` and remember a
+> few things so that later we can send our changes to `origin` with `git pull` without any
+> additional arguments.
```

Comment:
I do think it is helpful to have a short sentence saying that `-u` is equivalent to `--set-upstream-to`. It helps learners understand and remember why the flag is `-u`.

Reply:
The `--set-upstream-to` flag is a flag for the `git branch` command. We don't cover what `git branch` command does, so it is probably better to not mention it at all.
`-u` is a short form of `--set-upstream`, which we can mention.

Reply:
Yes! Sorry, I get those mixed up. I think it'd be worth mentioning `--set-upstream` in this text somewhere.

Reply:
Would something like this work?

 > While the `-u` flag is cryptic, its full name -- `--set-upstream` -- is more revealing.            
 > When `git push` is invoked with either of these two flags, Git updates branch `master` in the remote repository `origin` and sets that branch as an _upstream branch_ for the local branch `master`.                                                                                          
 > When a local branch "knows" its _upstream_ branch, we can move the changes between the two (pull and push) without specifying either the remote repository (_e.g._, `origin`) or the remote branch (_e.g._, `master`).

Reply:
I really love that explanation. +1 from me

Reply:
Thanks, Madicken. Do you want me to replace the `This option instructs Git` sentence or do you want me to append this text?

Reply:
Please let me know what you think about new version of the paragraph.